plat/common/include/x86/acpi/madt.h

```diff
@@ -0,0 +1,78 @@
+/* SPDX-License-Identifier: ISC */
+/*
+ * Authors: Cristian Vijelie <cristianvijelie@gmail.com>
+ *
+ * Copyright (c) 2021, University Politehnica of Bucharest. All rights reserved.
+ *
+ * Permission to use, copy, modify, and/or distribute this software
+ * for any purpose with or without fee is hereby granted, provided
+ * that the above copyright notice and this permission notice appear
+ * in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+ * CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS
+ * OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+ * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef __MADT_H__
+#define __MADT_H__
+
+#include <uk/arch/types.h>
+#include <x86/acpi/rsdt.h>
+
+struct MADT {
+	struct ACPISDTHeader h;
+	__u32 LocalAPICAddress;
+	__u32 Flags;
+	__u8 Entries[];
+};
+
+struct MADTEntryHeader {
+	__u8 Type;
+	__u8 Length;
+};
+
+struct MADTType0Entry {
+	struct MADTEntryHeader eh;
+	__u8 ACPIProcessorID;
+	__u8 APICID;
+	__u32 Flags;
+};
+
+struct MADTType1Entry {
+	struct MADTEntryHeader eh;
+	__u8 IOAPICID;
+	__u8 Reserved;
+	__u32 IOAPICAddress;
+	__u32 GlobalSystemInterruptBase;
+};
+
+struct MADTType2Entry {
+	struct MADTEntryHeader eh;
+	__u8 BusSource;
+	__u8 IRQSource;
+	__u32 GlobalSystemInterrupt;
+	__u16 Flags;
+};
+
+struct MADTType4Entry {
+	struct MADTEntryHeader eh;
+	__u8 ACPIProcessorID;
+	__u16 Flags;
+	__u8 LINT;
+};
+
+struct MADTType5Entry {
+	struct MADTEntryHeader eh;
+	__u16 Reserved;
+	__u64 LAPICOverride;
+};
+
+void find_madt(void);
```

Comment:
Maybe add a description comment for this function (e.g., that is searches within the first bytes of the address space). In general: Do you want to return a status code to indicate if you found it?